packages/functions/src/coordinator/CoordinatorBookAppointmentHandler.ts

```diff
@@ -0,0 +1,102 @@
+import { getDonorOrThrow, setDonor } from "../dal/DonorDataAccessLayer";
+import {
+  getAppointmentsByDonorIdInTime,
+  getAppointmentsByIds,
+  setAppointment,
+} from "../dal/AppointmentDataAccessLayer";
+import * as admin from "firebase-admin";
+import {
+  AppointmentStatus,
+  BookingChange,
+  FunctionsApi,
+  Hospital,
+  MANUAL_DONOR_ID,
+} from "@zm-blood-components/common";
+import { dbAppointmentToBookedAppointmentApiEntry } from "../utils/ApiEntriesConversionUtils";
+import { notifyOnAppointmentBooked } from "../notifications/BookAppointmentNotifier";
+import { BookAppointmentStatus } from "../../../common/src/functions-api";
+import { DbDonor } from "../function-types";
+
+const WEEKS_BUFFER = 0;
+
+export default async function (
```

Comment:
How much of this code is duplicate from the BookAppointmentHandler?

Can you extract common code to utils and prevent duplications?

Reply:
moved to a common file. I wondered what to do with the tests..

Reply:
done